docs/tasks/debug-application-cluster/logging-stackdriver.md

```diff
@@ -5,16 +5,94 @@ assignees:
+Before reading this page, it's highly recommended to familiarize yourself
+with the [overview of logging in Kubernetes](/docs/concepts/cluster-administration/logging).
 
-This article assumes that you have created a Kubernetes cluster with cluster-level logging support for sending logs to Stackdriver Logging. You can do this either by selecting the  **Enable Stackdriver Logging** checkbox in the create cluster dialogue in [GKE](https://cloud.google.com/container-engine/), or by setting the `KUBE_LOGGING_DESTINATION` flag to `gcp` when manually starting a cluster using `kube-up.sh`.
+**Note:** By default, only container's standard output and standard error will be
+collected by logging solution. To collect logs written by an application to
+a file, you can take a look at the [sidecar approach](/docs/concepts/cluster-administration/logging/#using-a-sidecar-container-with-the-logging-agent).
 
-The following guide describes gathering a container's standard output and standard error. To gather logs written by an application to a file, you can use [a sidecar approach](https://github.com/kubernetes/contrib/blob/master/logging/fluentd-sidecar-gcp/README.md).
+## Deploying
 
-## Overview
+In order to ingest logs, as explained in the logging overview, logging agent
+should be deployed to each node. Stackdriver logging agent is configured
+fluentd instance. Configuration for fluentd is stored in ConfigMap and
+instances are managed using DaemonSet, which makes Kubernetes schedule
+an agent on each node. The way how ConfigMap and DaemonSet are deployed
+in a cluster depends on your setup.
 
-After creation, you can discover logging agent pods in the `kube-system` namespace,
-one per node, by running the following command:
+### GKE
+
+Stackdriver is a default logging solution for clusters on GKE. Unless you
+deliberately opt out of using Stackdriver Logging, it will be deployed
+and working out of the box without any further actions.
+
+### Deploying with a new cluster
+
+**Note:** this section is only valid if you're using `kube-up.sh` for starting
+your cluster. Otherwise, please refer to the next section.
+
+Before starting your cluster, set `KUBE_LOGGING_DESTINATION` environment
+variable to `gcp` and, unless on GCE, include
+`alpha.kubernetes.io/fluentd-ds-ready=true` to`KUBE_NODE_LABELS` variable.
+Once started, cluster should have Stackdriver logging agent running on each
+node. If deployed that way, DaemonSet for the agents and ConfigMap with
+the agents configuration will be addons.
+
+### Deploying to an existing cluster
+
+1. Apply label on each node, if not present already.
+
+    Stackdriver logging agent deployment is using node labels to understand to
+    which nodes it should be allocated. If describing your nodes produces the
+    following output
+
+    ```shell
+    $ kubectl describe node $NODE_NAME
+    ...
```

Comment:
Oh. Sorry. That's our standard style--we separate commands and responses. So it should be like this:

"Run this command:"

    kubectl command

"The output is similar to this:"

    console output